### PR TITLE
Integrate `ModularTokenizerOp` with Hugging Face remote 🤗

### PR DESCRIPTION
**Pull request by SagiPolaczek**
Supporting HF-like API to integrate the modular_tokenizer_op with HF remote. 

API:
```python

from fuse.data.tokenizers.modular_tokenizer.op import ModularTokenizerOp

tokenizer_op = ModularTokenizerOp.from_pretrained("repo/id")
# OR 
tokenizer_op = ModularTokenizerOp.from_pretrained("/local/path/to/tokenizer/dir")
```

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -58,7 +58,7 @@ exclude =
 
 
 [mypy]
-python_version = 3.7
+python_version = 3.9
```
Comment (SagiPolaczek) [inline setup.cfg:61]: due `mypy` pre-commit errors

File: fuse/data/tokenizers/modular_tokenizer/op.py [hunk 2]
```diff
@@ -506,3 +509,83 @@ def __call__(
...
+                ) from e
+
+        tokenizer_op = cls(
+            tokenizer_path=identifier, pad_token=pad_token, max_size=max_size
+        )
+        return tokenizer_op
+
+    def save_pretrained(self, save_directory: Union[str, Path]) -> None:
+        print(f"Saving @ {save_directory=}")
+        self._tokenizer.save(path=str(save_directory))
```
Comment (SagiPolaczek) [inline fuse/data/tokenizers/modular_tokenizer/op.py:556]: As far as I understand, all of the information is being stored in `tokenizer`. That's why it's enough to save it.

Comment (mosheraboh) [reply fuse/data/tokenizers/modular_tokenizer/op.py:556]: Correct

**Review by mosheraboh — DISMISSED** [review]
LGTM
Added inline suggestion to  consider.

**Pull request merged**